### PR TITLE
main

### PR DESCRIPTION
**Pull request by jnumainville**
Depends on https://github.com/deephaven/web-client-ui/issues/1373

Added support for pydeephaven tables

Tested with the following code in a jupyter notebook (with appropriate connection_info, user and password)
```
import datetime
import time
from deephaven_enterprise.client.session_manager import SessionManager
from deephaven_ipywidgets import DeephavenWidget

connection_info = "connection"
session_mgr: SessionManager = SessionManager(connection_info)
session_mgr.password("user", "password")

session = session_mgr.connect_to_persistent_query("Test PQ")
qc = session.open_table("tt")
print("Table size: ", qc.size)
```
```
DeephavenWidget(qc)
```
```
session = session_mgr.connect_to_new_worker(name=None, heap_size_gb=1.0)
# We can run arbitrary script code
session.run_script('from deephaven import time_table\n'
                   'tt3 = time_table("PT3s").update("Remote=false")')

tt = session.open_table("tt3")
```
```
DeephavenWidget(tt)
```
```
session = session_mgr.connect_to_new_worker(name=None, heap_size_gb=1.0)
# Or do table based operations
tt = session.time_table(period=1000_000_000).update("Remote=true")
session.bind_table("tt", tt)
print(tt.to_arrow())
```
```
DeephavenWidget(tt)
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
   },
   "dependencies": {
     "@deephaven/jsapi-shim": "0.15.0",
+    "@deephaven/jsapi-utils": "^0.41.0",
```
Comment (mofojed) [inline package.json:54]: We should update the other `@deephaven/*` packages as well

**Review by mofojed — CHANGES_REQUESTED** [review]

File: src/widget.ts [hunk 2]
```diff
@@ -51,11 +56,64 @@ export class DeephavenModel extends DOMWidgetModel {
...
+    }
+
+    if (!isMessage(data)) {
+      log.debug('Ignore unsupported message', data);
+      return;
+    }
+
+    switch (data.message) {
+      case LOGIN_OPTIONS_REQUEST:
+        window.removeEventListener('message', this.handleAuthentication);
```
[the remark this answers is not in the document]
Comment (jnumainville) [reply src/widget.ts:100]: It seems like there is `observe` to watch for changes to traitlets that could be used to communicate if a new token is needed. A couple difficulties: 
1. What's a good way to bind listeners to specific object_ids to make sure not every listener is being used on every object?
2. It seems like, on iframe reload, I would need to somehow prevent the reload from going through until the new token is retrieved from the backend. What's a good way to do that?

File: package.json (hunk 1 above)
Comment (mofojed) [reply package.json:54]: @jnumainville I've published v0.41.3 which you can use now

File: src/widget.ts [hunk 3]
```diff
@@ -51,11 +56,64 @@ export class DeephavenModel extends DOMWidgetModel {
 export class DeephavenView extends DOMWidgetView {
   private iframe: HTMLIFrameElement;
 
+  sendAuthenticationResponse = (
+    messageId: string,
+    source: WindowProxy,
```
Comment (mofojed) [inline src/widget.ts:61]: In the context of this function, `source` doesn't really make sense. `source` is the source of the event, but in the context of this function, we're just passing a function to a child window. I'd suggest changing the parameter name to `childWindow` here (or something similar).

File: src/widget.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mofojed) [reply src/widget.ts:100]: @jnumainville to your questions above:
1. I think what you're asking is how does the parent window know the request is coming from it's specific iframe? In that case you can check the `source` from the event. I think a check `source === this.iframe` should work.
2. We can do this as a follow up; but I imagine the best way to do that would be to not create the iframe until we've got the auth token from python... so perhaps have a value for `authIteration`, increment that each time the JS loads, python plugin picks up that the value has updated and retrieves a new auth token, sets that as another traitlet value that UI picks up on.
This blog has a decent example: https://blog.jupyter.org/authoring-custom-jupyter-widgets-2884a462e724
I think you should be able to do it like that?

**Review by mofojed — CHANGES_REQUESTED** [review]
Looks good, just need to update to fix the build/test issue that's occurring in CI: https://github.com/deephaven/deephaven-ipywidgets/actions/runs/5339190350/jobs/9677572562?pr=19
Likely need to include the peerDependencies as `devDependencies` in this project: https://github.com/deephaven/web-client-ui/blob/e23695d4baf232720ca89cb7d24e9a918f3fe913/packages/eslint-config/package.json#L24

**Comment by jnumainville** [comment]
@mofojed just saw your comment... do I need exactly version 0.3.4? or is this new error something else entirely?

**Comment by mofojed** [comment]
The error in the build is:
```
Error: Error while loading rule '@typescript-eslint/strict-boolean-expressions': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
```

I think you just need to add this to the `.eslintrc`, similar to web-client-ui: https://github.com/deephaven/web-client-ui/blob/e23695d4baf232720ca89cb7d24e9a918f3fe913/.eslintrc.js#L8
```
      parserOptions: {
        project: ['./tsconfig.eslint.json', './packages/*/tsconfig.json'],
        tsconfigRootDir: __dirname,
      },
```

**Review by mofojed — APPROVED** [review]

**Pull request merged**